### ISSUE TITLE
Fish function are not completions

### ISSUE THREAD

**Issue by Zykino** [zellij-org/zellij#2186]
Doing `zellij setup --generate-completion fish > ~/.config/fish/completions/zellij.fish` create a lot of nice completions around zellij.

I have 2 issues with it:
The completions files are evaluated lazily but some functions are in there `ze` `zrf`… They should be in `~/.config/fish/functions/zellij.fish` where they will be evaluated more eagerly (so I can use `ze` without having to do `zellij [Tab]` first.
=> We need 2 generation commands or 2 outputs. This can be worked around and fixed manually… if the users know all that :sweat_smile:.
I’m not sure how other shell’s completions are working.

`zellij run [Tab]` (and `zellij action new-pane [Tab]`, maybe others too) try to complete with local folder instead of commands. I’m not sure it can be addressed, I do not know the fish completion ^^'.

**Comment by Zykino** [comment]
**Note:** As a user, to save the functions just do:
```sh
zellij setup --generate-completion fish > ~/.config/fish/completions/zellij.fish
zellij [Tab] [Ctrl+c]
funcsave zr
funcsave zrf
funcsave ze
funcsave zef
```

This will save each function into its own autoloaded function file in `~/.config/fish/functions/z*`.

**Comment by imsnif** [comment]
Yeah, I'll readily admit I kind of hacked these together in the last minute before the release. So you're suggesting adding a `--generate-shell-aliases` flag instead? Or some other name?

**Comment by Zykino** [comment]
I don’t really know. Another possibility is to generate all files and directory structure (possibly in `/tmp/zellijConfig`). And in stdout tell the user to inspect it and then `mv /tmp/zellijConfig ~/.config/fish` if they are ok to keep it. I mean, if you do not want to overwrite default config automatically.

I’m not sure generating multiples aliases is possible in a single file, the way fish works. Or maybe `ze` and `zef` in `ze.fish` and the same for the `zr` family? I must admit, I use fish but am not well versed into its configs.

**Comment by Zykino** [comment]
`skim` does not propose auto-installing their function (or I’m not aware how). They propose this file: https://github.com/lotabout/skim/blob/master/shell/key-bindings.fish. I installed it with:
```
curl https://raw.githubusercontent.com/lotabout/skim/master/shell/key-bindings.fish > ~/.config/fish/functions/skim.fish
< edit the file to insert the command as a reminder ;) >
```
zellij can offer the same and all the aliases in the same file: zellij.fish.

Thinking again, maybe this can be a good citizen default: not "steal" all the short name and expose the quick functions as `zellije`, `zellijef`, `zellijr`, `zellijrf`. Easier to remember/learn for occasional user.

Maybe on the "shell integration" doc can propose to do the `abbr -a -U -- ze zellije`.

**Comment by jirijakes** [comment]
I don't know what would be the fishiest way but one more option how to make the completions and aliases work now is adding:

```zellij setup --generate-completion fish | source```

into `config.fish`. It's probably not the most idiomatic way but AFAIK it's also what zoxide and starship recommend.

**Comment by Zykino** [comment]
Effectively `starship` and `skim` ask to do something like this in the `config.fish`, inside the `if status is-interactive` block.
```fish
if status is-interactive
	# Commands to run in interactive sessions can go here

	# Skim key bindings
	source ~/.config/fish/functions/skim.fish
	skim_key_bindings

	starship init fish | source
end
```

So this is a possibility, but I prefer `skim`'s setup: do not generate the config each time. Especially for the completion: as a user you prefer them not to change without your knowledge.

**Comment by Zykino** [comment]
I saw #2650 is trying to fix this :)

**Comment by aldanor** [comment]
Another thing I noticed (unless it has been already discussed somewhere), in the fish shell,

- `ze` function auto-completes file paths
- `zef` function does *not* autocomplete file paths which renders it next to useless (it tries to auto-complete options but never file paths)

**Comment by upryst** [comment]
Struggling with `zef` completion behaviour as well. It looks like there's some collision with `zef` from Raku universe, which [Fish is trying to support out of the box](https://github.com/fish-shell/fish-shell/blob/master/share/completions/zef.fish).
